components/daikin_s21/s21.h

```diff
@@ -25,6 +26,217 @@ enum class DaikinFanMode : uint8_t {
+  inline constexpr const char* OptionalFeatures = "G2";
+  inline constexpr const char* OnOffTimer = "G3";
+  inline constexpr const char* ErrorStatus = "G4";
+  inline constexpr const char* SwingOrHumidity = "G5"; //
+  inline constexpr const char* SpecialModes = "G6"; //
+  inline constexpr const char* DemandAndEcono = "G7"; //
+  inline constexpr const char* OldProtocol = "G8"; //
+  inline constexpr const char* InsideOutsideTemperatures = "G9"; //
+  // GA
+  // GB
+  inline constexpr const char* ModelCode = "GC";
+  inline constexpr const char* IRCounter = "GG";
+  inline constexpr const char* V2OptionalFeatures = "GK";
+  // GL
+  inline constexpr const char* PowerConsumption = "GM";
+  // GN
+  // GP
+  // GQ
+  inline constexpr const char* LouvreAngle = "GR";
+  // GS
+  // GT
+  inline constexpr const char* V3OptionalFeatures = "GU00";
+  inline constexpr const char* AllowedTemperatureRange = "GU02";
+  // GU04
+  inline constexpr const char* ModelName = "GU05";
+  inline constexpr const char* ProductionInformation = "GU15";
+  inline constexpr const char* ProductionOrder = "GU25";
+  inline constexpr const char* IndoorProductionInformation = "GU35";
+  inline constexpr const char* OutdoorProductionInformation = "GU45";
+  // GV
+  // GX00
+	// GX10
+	// GX20
+	// GX30
+	// GX40
+	// GX50
+	// GX60
+	// GX70
+	// GX80
+	// GX90
+	// GXA0
+	// GXB0
+	// GXC0
+	// GXD0
+	// GXE0
+	// GXF0
+	// GX01
+	// GX11
+	// GX21
+	// GX31
+	// GX41
+	// GX51
+	// GX61
+	// GX71
+	// GX81
+  inline constexpr const char* NewProtocol = "GY00"; //
+  // GY10
+  // GY20
+}  // namespace StateResponse
+
+namespace EnvironmentResponse {
+  inline constexpr const char* PowerOnOff = "SA";
+  inline constexpr const char* IndoorUnitMode = "SB";
+  inline constexpr const char* TemperatureSetPoint = "SC";
+  inline constexpr const char* OnTimerSetting = "SD";
+  inline constexpr const char* OffTimerSetting = "SE";
+  inline constexpr const char* FanMode = "SG";
+  inline constexpr const char* InsideTemperature = "SH";
+  inline constexpr const char* LiquidTemperature = "SI";
+  inline constexpr const char* FanSetPoint = "SK";
+  inline constexpr const char* FanSpeed = "SL";
+  inline constexpr const char* LouvreAngleSetPoint = "SM";
+  inline constexpr const char* VerticalSwingAngle = "SN";
+  // SW
+  inline constexpr const char* TargetTemperature = "SX"; // (not set point, see details)
+  inline constexpr const char* OutsideTemperature = "Sa";
+  inline constexpr const char* IndoorFrequencyCommandSignal = "Sb";
+  inline constexpr const char* CompressorFrequency = "Sd";
+  inline constexpr const char* IndoorHumidity = "Se";
+  inline constexpr const char* CompressorOnOff = "Sg";
+  inline constexpr const char* UnitState = "SzB2";
+  inline constexpr const char* SystemState = "SzC3";
+  // Sz52
+  // Sz72
+} // namespace EnvironmentResponse
+
+namespace OldProtocol {
+  inline constexpr const char* Protocol0 = "\x30\x00\x00\x00";
```

Comment:
There are null bytes in this string literal. I prefer bounded types for these responses instead of implicit knowledge that they have 4 valid bytes available to read. An enumerated protocol version type is probably more useful than my major/minor type though -- easier to feed into a switch statement. I think I'll go with that in a future PR and a HumanReadableProtocol string lookup.

Reply:
Yes, I forgot to make note of that when diffing our codebases. Every query is equal at the moment, though new commands will preempt the query cycle. I figured that since some settings changes take a few seconds to be reflected anyways, it wouldn't be a big deal, but if there's a use case for a particular set of sensor being polled more often then I'm open to it. We could draw a better swing angle sine wave in the logbook, but that doesn't seem too worthwhile.

Reply:
I think using numeric comparisons for protocol version support would probably be better than enums, to more gracefully support undocumented versions. If you store protocol v0 as `0`, v2 as `200`, v3.20 as `320` and so on, and you end up with some v3.60 protocol unit, you can still expose the full feature set of 3.20 or 3.40 instead of falling back to a minimum core set for unknown protocol versions. In fact, that's probably why they moved from F8 to FY00 starting with v3, when they realized that whatever matching method they were previously using wasn't flexible enough for future-proof protocol extension.